src/main/java/io/reactivex/observers/TestObserver.java

```diff
@@ -255,7 +255,8 @@ public final boolean hasSubscription() {
      */
     @Override
     public final TestObserver<T> assertNotSubscribed() {
-        if (subscription.get() != null) {
+        Disposable disposable = subscription.get();
+        if (disposable != null && disposable != DisposableHelper.DISPOSED) {
```

Comment:
If this gets changed [TestSubscriber](https://github.com/ReactiveX/RxJava/blob/ff282b9647d0accc0cb55780c9bdb97f4ed03053/src/main/java/io/reactivex/subscribers/TestSubscriber.java#L316) should be changed too for consistency